### PR TITLE
E3 agg commitment

### PR DESCRIPTION
**Pull request by awskii**

### CODE REVIEW

File: state/history.go [hunk 1]
```diff
@@ -30,21 +30,22 @@ import (
...
 
 type History struct {
 	*InvertedIndex
 	historyValsTable string // key1+key2+txnNum -> oldValue , stores values BEFORE change
 	settingsTable    string
-
-	files        *btree.BTreeG[*filesItem]
-	compressVals bool
+	files            *btree.BTreeG[*filesItem]
+	valueMergeFn     func(prev, last []byte) ([]byte, error)
```
Comment (AlexeyAkhunov) [inline state/history.go:47]: Perhaps this valueMergeFn should belong to `Domain` instead? I did not see how it is used in `History`

File: recsplit/eliasfano32/elias_fano.go [hunk 2]
```diff
@@ -305,8 +305,12 @@ func ReadEliasFano(r []byte) (*EliasFano, int) {
 	ef.count = binary.BigEndian.Uint64(r[:8])
 	ef.u = binary.BigEndian.Uint64(r[8:16])
 	ef.maxOffset = ef.u - 1
-	p := (*[maxDataSize / 8]uint64)(unsafe.Pointer(&r[16]))
-	ef.data = p[:]
+
+	ef.data = make([]uint64, len(r[16:])/8)
```
Comment (AlexeyAkhunov) [inline recsplit/eliasfano32/elias_fano.go:309]: This needs to be undone, because the idea is to use memory-mapped files and not allocate it on the heap

Comment (awskii) [reply recsplit/eliasfano32/elias_fano.go]: we discussed that already, using unsafe pointer `p` leads to use-after-close in indexes

**Comment by awskii** [comment]
Fuzz test requires to be revisited, everything else looks working.

**Pull request merged**